pandablocks/utils.py

```diff
@@ -56,18 +61,23 @@ def words_to_table(
         if field_info.subtype == "int":
             # First convert from 2's complement to offset, then add in offset.
             packing_value = (value ^ (1 << (bit_length - 1))) + (-1 << (bit_length - 1))
-        elif field_info.labels:
+        elif convert_enum_indices and field_info.labels:
             packing_value = [field_info.labels[x] for x in value]
         else:
-            packing_value = value
+            if bit_length <= 8:
+                packing_value = value.astype(np.uint8)
+            elif bit_length <= 16:
+                packing_value = value.astype(np.uint16)
+            else:
+                packing_value = value.astype(np.uint32)
 
         unpacked.update({field_name: packing_value})
 
     return unpacked
 
 
 def table_to_words(
-    table: Dict[str, Iterable], table_field_info: TableFieldInfo
+    table: Dict[str, Union[np.ndarray, List]], table_field_info: TableFieldInfo
```

Comment:
I believe the type hint you want here is `Sequence` - that is both an `Iterable` and has `Sized`.

Alternatively, this should be `UnpackedArray`, to indicate it's the same return type as what `words_to_table` outputs.